Source/WebCore/platform/ios/LegacyTileCache.mm

```diff
@@ -586,7 +586,7 @@ static bool shouldRepaintInPieces(const CGRect& dirtyRect, CGSRegionObj dirtyReg
         CFTypeRef keys[] = { kCTFontAttributeName, kCTForegroundColorFromContextAttributeName };
         CFTypeRef values[] = { font.get(), kCFBooleanTrue };
         auto attributes = adoptCF(CFDictionaryCreate(kCFAllocatorDefault, keys, values, std::size(keys), &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks));
-        auto string = adoptCF(CFStringCreateWithBytesNoCopy(kCFAllocatorDefault, reinterpret_cast<const UInt8*>(text), strlen(text), kCFStringEncodingUTF8, false, kCFAllocatorNull));
+        auto string = adoptCF(CFStringCreateWithCStringNoCopy(kCFAllocatorDefault, text, kCFStringEncodingUTF8, kCFAllocatorNull));
```

Comment:
I think there's a bug here. 'text' is not null terminated. I think you need to add a trailing \0 to the snprintf above.

Reply:
Well, strlen wasn't going to get that, but yeah.

Reply:
snprintf would null terminate unless the base-10 string fills the entire buffer, which means paint count would be 1,000,000,000,000,000, which is more than 32 bit paintcount can have.

Reply:
layer.paintCount is an unsigned integer.

Reply:
The bug is that this is %d, and not %u.

Reply:
I would prefer it if you fixed the bug. This call to CFStringCreateWithCStringNoCopy, while it has the potential to improve readability, is currently incorrect.

Reply:
So, yes snprintf always null terminates in this case.

Reply:
Hmmm... actually it looks like snprintf is specified to write a null terminator, as of C99: https://en.cppreference.com/w/c/io/fprintf.

Reply:
Actually, I have a better idea honestly.